faiss_vector_posting.go

```diff
@@ -335,9 +336,9 @@ func (sb *SegmentBase) InterpretVectorIndex(field string, except *roaring.Bitmap
 				return rv, nil
 			},
 			close: func() {
-				if vecIndex != nil {
-					vecIndex.Close()
-				}
+				// skipping the closing because the index is cached and it's being
+				// deferred to a later point of time.
+				sb.vectorCache.decRef(fieldIDPlus1)
 			},
 			size: func() uint64 {
 				if vecIndex != nil {
```

Comment:
Initialize a variable vecIndexSize at line 299 above.
Assign vecIndex.Size() to it at the time we load.
So size will simply return vecIndexSize